src/AppInstallerRepositoryCore/Microsoft/ARPHelper.cpp

```diff
@@ -3,11 +3,133 @@
 #include "pch.h"
 #include "ARPHelper.h"
 #include "winget/PortableARPEntry.h"
+#include <Msi.h>
 
 namespace AppInstaller::Repository::Microsoft
 {
     using namespace AppInstaller::Registry::Portable;
 
+    namespace
+    {
+        // "Unpacks" a GUID in the format used by the UpgradesCode registry key into the usual format.
+        // Returns empty if it is not a valid GUID
+        std::optional<std::string> TryUnpackUpgradeCodeGuid(std::string_view packed)

```

Comment:
As discussed, there is a simpler way to do this just using a transformation map from the given 32 character input into the desired GUID-like output.  Up to you on whether to change to that.

I do want to confirm that the output format here does match the value that would be returned by `MsiGetProperty`, as wingetcreate et al. will likely use that function to read the property out of the MSI files themselves when creating manifests.

Reply:
Changed to use the transformation map.

Yes, it does match.